src/runtime_src/xdp/profile/core/summary_writer.cpp

```diff
@@ -283,34 +300,73 @@ namespace xdp {
         rolloverCounts = mRolloverCountsMap.at(key);
       else
         memset(&rolloverCounts, 0, sizeof(xclCounterResults));
-      uint32_t  numHostSlots = mPluginHandle->getProfileNumberSlots(XCL_PERF_MON_HOST, deviceName);
-      for (uint32_t s=HostSlotIndex; s < HostSlotIndex + numHostSlots; s++) {
+
+      if (monitorType != RTUtil::MON_HOST_DYNAMIC) {
+        totalReadBytes    = 0;
+        totalWriteBytes   = 0;
+        totalReadTranx    = 0;
+        totalWriteTranx   = 0;
+        totalReadLatency  = 0;
+        totalWriteLatency = 0;
+      }
+
+      // Traverse all slots to find host monitors
+      uint32_t numSlots = mPluginHandle->getProfileNumberSlots(XCL_PERF_MON_MEMORY, deviceName);
+
+      for (uint32_t s=0; s < numSlots; s++) {
+        // Make sure it's a host monitor
+        uint32_t properties = mPluginHandle->getProfileSlotProperties(XCL_PERF_MON_MEMORY, deviceName, s);
+        if (!(properties & XSPM_HOST_PROPERTY_MASK))
+          continue;
+
+        std::string slotName;
+        mPluginHandle->getProfileSlotName(XCL_PERF_MON_MEMORY, deviceName, s, slotName);
+        if (slotName.find(monitorName) == std::string::npos)
+          continue;
+
         totalReadBytes += counterResults.ReadBytes[s]
                           + (rolloverCounts.ReadBytes[s] * 4294967296UL);
         totalWriteBytes += counterResults.WriteBytes[s]
                           + (rolloverCounts.WriteBytes[s] * 4294967296UL);
+        totalReadTranx += counterResults.ReadTranx[s]
+                          + (rolloverCounts.ReadTranx[s] * 4294967296UL);
+        totalWriteTranx += counterResults.WriteTranx[s]
```

Comment:
should we get rid of this? this is not valid now as we have 64 bit counters.

Reply:
Since the XDP library can be used outside of a controlled environment like SDAccel, it's possible for someone to insert monitors with only 32-bit counters. We'd have to support that. Let's keep it for now.